dll/win32/msafd/misc/sndrcv.c

```diff
@@ -687,38 +803,47 @@ WSPSend(SOCKET Handle,
                                     APCContext,
                                     IOSB,
                                     IOCTL_AFD_SEND,
-                                    &SendInfo,
-                                    sizeof(SendInfo),
+                                    SendInfo,
+                                    sizeof(AFD_SEND_INFO),
                                     NULL,
                                     0);
-
-    /* Wait for completion of not overlapped */
-    if (Status == STATUS_PENDING && lpOverlapped == NULL)
+    if (Status == STATUS_PENDING)
     {
-        WaitForSingleObject(SockEvent, INFINITE); // BUGBUG, shouldn wait infinitely for send...
+        /* Wait for completion of not overlapped */
+        if (!lpOverlapped)
+        {
+            /* BUGBUG, shouldn't wait infinitely for send... */
```

Comment:
```suggestion
            /* FIXME: shouldn't wait infinitely for send... */
```
same here and (in case there would be others that I missed) all other places.

Reply:
What would the timeout value be? Something finite but hardcoded? Something obtained from a setting from the registry? Something else?

Reply:
Not sure, but I think this can be fixed later.